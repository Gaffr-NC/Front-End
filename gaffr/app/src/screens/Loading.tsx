import React from 'react';
import {
  View,
  Text,
  ActivityIndicator,
  AsyncStorage,
  Image,
  StyleSheet
} from 'react-native';
import { NavigationScreenProp } from 'react-navigation';

interface Props {
  navigation: NavigationScreenProp<any, any>;
}

export default class Loading extends React.Component<Props> {
  async componentDidMount() {
    const uid = await AsyncStorage.getItem('uid');
    const userType = await AsyncStorage.getItem('userType');
    if (uid && userType) {
      this.props.navigation.navigate(
        userType === 'tenants' ? 'Swiper' : 'Properties'
      );
    } else {
      AsyncStorage.removeItem('userType');
      AsyncStorage.removeItem('uid');
      this.props.navigation.navigate('logIn');
    }
  }
  render() {
    return (
      <View style={styles.loadingContainer}>
        <Image
          style={styles.logo}
          source={{ uri: 'https://i.imgur.com/NH0xhhe.png' }}
        />
        <Text style={styles.headerText}>
          It's time to move forward, and it's time to move in...
        </Text>
        <ActivityIndicator size="large" />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  loadingContainer: {
    backgroundColor: '#dcd1e8',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  logo: {
    alignItems: 'center',
    height: 150,
    width: 200
  },
  headerText: {
    textAlign: 'center',
    alignItems: 'center',
    fontSize: 18,
    padding: 0
  }
});
