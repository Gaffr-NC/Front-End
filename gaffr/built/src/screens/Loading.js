"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const react_native_1 = require("react-native");
class Loading extends react_1.default.Component {
    render() {
        return (react_1.default.createElement(react_native_1.View, { style: { flex: 1, alignItems: 'center', justifyContent: 'center' } },
            react_1.default.createElement(react_native_1.Text, null, "Loading..."),
            react_1.default.createElement(react_native_1.ActivityIndicator, { size: "large" })));
    }
}
exports.default = Loading;
