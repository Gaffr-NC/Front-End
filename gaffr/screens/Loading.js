import React, { Component } from "react";
import { View, Text, ActivityIndicator } from "react-native";
import firebase from 'react-native-firebase';

export default class Loading extends Component {
  render() {
    return (
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Text>Loading...</Text>
        <ActivityIndicator size="large" />
      </View>
    );
  }

  componentDidMount() {
    firebase.auth().onAuthStateChanged(user => {
      this.props.navigation.navigate(user ? 'TenantApp' : 'Auth')
    })
  }
}