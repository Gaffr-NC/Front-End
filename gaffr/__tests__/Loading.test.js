import React from "react";
import renderer from "react-test-renderer";
import Loading from "../screens/Loading.js";

it("Loading renders without crashing", () => {
  const rendered = renderer.create(<Loading />).toJSON();
  expect(rendered).toBeTruthy();
});
it("Tests Loading against snapshot", () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
