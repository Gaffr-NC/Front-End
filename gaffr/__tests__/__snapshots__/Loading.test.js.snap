// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Loading against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Loading...
  </Text>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Tests LoginScreen against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Log in or sign up!
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    label="email"
    onChangeText={[Function]}
    placeholder="email..."
    style={
      Object {
        "backgroundColor": "powderblue",
        "margin": 10,
        "padding": 10,
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    label="password"
    onChangeText={[Function]}
    placeholder="password..."
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "powderblue",
        "margin": 10,
        "padding": 10,
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        LOG IN
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        SIGN UP
      </Text>
    </View>
  </View>
</View>
`;
